pandas/core/nanops.py

```diff
@@ -1039,7 +1039,20 @@ def reduction(
             except (AttributeError, TypeError, ValueError):
                 result = np.nan
         else:
-            result = getattr(values, meth)(axis)
+            try:
+                result = getattr(values, meth)(axis)
+            except TypeError as e:
+                # the only case when this can happein is for timezone aware
+                # Timestamps  where a NaT value is casted to # np.inf
+                from pandas.core.dtypes.common import is_float
+
+                vfunc = np.vectorize(lambda x: is_float(x))
```

Comment:
i think we'd be better off fixing _get_values so as to not incorrectly insert a float